front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx

```diff
@@ -30,25 +50,143 @@ export default function ImportTrainScheduleConfig(props: ImportTrainScheduleConf
   const dispatch = useDispatch();
   const { openModal, closeModal } = useContext(ModalContext);
 
+  async function importTrackSections(opIds: number[]): Promise<Record<number, TrackLocation[]>> {
+    const uniqueOpIds = Array.from(new Set(opIds));
+    const constraint = uniqueOpIds.reduce((res, uic) => [...res, ['=', ['uic'], Number(uic)]], [
+      'or',
+    ] as (string | SearchConstraintType)[]);
+    const payload: PostSearchApiArg = {
+      body: {
+        object: 'operationalpoint',
+        query: ['and', constraint, ['=', ['infra_id'], infraId]],
+      },
+      pageSize: 1000,
+    };
+    const operationalPoints = (await postSearch(
+      payload
+    ).unwrap()) as SearchOperationalPointResult[];
+
+    return operationalPoints.reduce(
+      (res, operationalPoint) =>
+        operationalPoint.uic
+          ? {
+              ...res,
+              [operationalPoint.uic]: operationalPoint.track_sections,
+            }
+          : res,
+      {}
+    );
+  }
+
+  function validateImportedTrainSchedules(
+    importedTrainSchedules: Record<string, unknown>[]
+  ): ImportedTrainSchedule[] | null {
+    const isInvalidTrainSchedules = some(importedTrainSchedules, (trainSchedule) => {
+      if (
+        !trainSchedule.trainNumber ||
+        !trainSchedule.rollingStock ||
+        !trainSchedule.departureTime ||
+        !trainSchedule.arrivalTime ||
+        !trainSchedule.departure ||
+        !trainSchedule.steps ||
+        !Array.isArray(trainSchedule.steps)
+      ) {
+        return true;
+      }
+      const hasInvalidateSteps = some(
+        trainSchedule.steps,
+        (step) =>
+          !('arrivalTime' in step) ||
+          !('departureTime' in step) ||
+          !('uic' in step) ||
+          // !('yard' in step) ||
+          !('name' in step) ||
+          !('trigram' in step) ||
+          !('latitude' in step) ||
+          !('longitude' in step)
+      );
```

Comment:
```suggestion
        (step) => ['arrivalTime', 'departureTime', …].some(key => !stop.hasOwnProperty(key))
      );
```

`in` checks the entire prototype chain
`hasOwnProperty` only checks into the object

why do you check if the key exists rather than check if the value is defined ?

Reply:
Because some of the strings can be empty ('') or some of the value can be 0 etc, so that's why I decided to check only if the key exists !

Reply:
Done, I used 'in'

Reply:
ok then use `hasOwnProperty` (prototype chain comment above)